neo4j/src/main/java/org/apache/metamodel/neo4j/Neo4jRequestWrapper.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.neo4j;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import org.apache.http.HttpHost;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.BaseEncoding;
+
+/**
+ * The class takes care of sending an {@link HttpRequestBase} or a Cypher query
+ * to the specified Neo4j instance. Also takes care of the authentication.
+ * 
+ */
+public class Neo4jRequestWrapper {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(Neo4jRequestWrapper.class);
+
+	private final CloseableHttpClient _httpClient;
+	private final HttpHost _httpHost;
+	private final HttpPost _cypherQueryHttpPost;
+	private final String _username;
+	private final String _password;
+
+	public Neo4jRequestWrapper(CloseableHttpClient httpClient,
+			HttpHost httpHost, String username, String password) {
+		_httpClient = httpClient;
+		_httpHost = httpHost;
+		_username = username;
+		_password = password;
+		_cypherQueryHttpPost = new HttpPost("/db/data/transaction/commit");
```

Comment:
I am unsure how this code (and the dataset that eventually consumes it) will work on queries with large results. Is there not some kind of paging mechanism in Neo4j so that you should fire another request once you've traversed the first? If not all records have to fit into the response JSON which seems very memory intensive and non-scalable.